lib/pure/osproc.nim

```diff
@@ -410,6 +410,45 @@ when defined(Windows) and not defined(useNimRtl):
       copyMem(addr(result[L]), cstring(x), x.len+1) # copy \0
       inc(L, x.len+1)
 
+  proc envToWideCString(env: StringTableRef): WideCString =
+    # newWideCString stops on \0 characters, so we have to
+    # convert every pair separately.
+    const wcharSize = 2
+    var rows = newSeq[tuple[str: WideCString, len: int]]()
+    var L = 0  # length in wide chars
+
+    for key, val in pairs(env):
+      let str = newWideCString(key & "=" & val)
+      let row = (str: str, len: str.len)
+      rows.add(row)
+      inc(L, succ row.len)
+
+    # Leave space for trailing \0
+    result = cast[WideCString](alloc0(wcharSize * (succ L)))
```

Comment:
My bad. Used the existing code (`envToCString`, former `buildEnv`) as a template, didn't look closely at types... Will fix first thing tomorrow.